sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java

```diff
@@ -63,23 +64,27 @@ public static synchronized void getDbToken(
             TokenExchangeServiceCallback serviceCallback) {
         AppCenterLog.debug(LOG_TAG, "Getting a resource token from App Center...");
         String url = apiUrl + GET_TOKEN_PATH_FORMAT;
-
-        ServiceCall tokenResponse =
-            httpClient.callAsync(
-                url,
-                METHOD_POST,
-                new HashMap<String, String>() { { put(APP_SECRET_HEADER, appSecret); } },
-                    new HttpClient.CallTemplate() {
-
-                        @Override
-                        public String buildRequestBody() {
-                            return buildAppCenterGetDbTokenBodyPayload(partition);
-                        }
-
-                        @Override
-                        public void onBeforeCalling(URL url, Map<String, String> headers) { }
-                    },
-                    serviceCallback);
+        TokenResult tokenResult = TokenManager.getInstance().getToken(partition);
```

Comment:
Are we caching by partition? I thought it was by documentId. We also discussed adding a userId in the string potentially in the future.

Reply:
Also, let's call the variable something like `cachedToken` for better readability

Reply:
hmmm, I am not sure why should be partitioned by documentId. If we get the token by partition, it would be able to read all the document under the partition...